x-pack/plugins/siem/server/lib/ip_details/query_users.dsl.ts

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { Direction, UsersFields, UsersSortField } from '../../graphql/types';
+import { assertUnreachable, createQueryFilterClauses } from '../../utils/build_query';
+
+import { UsersRequestOptions } from './index';
+
+export const buildUsersQuery = ({
+  ip,
+  usersSortField,
+  filterQuery,
+  flowTarget,
+  pagination: { limit },
+  sourceConfiguration: {
+    fields: { timestamp },
+    auditbeatAlias,
+    packetbeatAlias,
+    winlogbeatAlias,
+  },
+  timerange: { from, to },
+}: UsersRequestOptions) => {
+  const filter = [
+    ...createQueryFilterClauses(filterQuery),
+    { range: { [timestamp]: { gte: from, lte: to } } },
+    { term: { [`${flowTarget}.ip`]: ip } },
+  ];
+
+  const dslQuery = {
+    allowNoIndices: true,
+    index: [auditbeatAlias, packetbeatAlias, winlogbeatAlias],
+    ignoreUnavailable: true,
+    body: {
+      aggs: {
+        user_count: {
+          cardinality: {
+            field: 'user.name',
+          },
+        },
+        users: {
+          terms: {
+            field: 'user.name',
+            size: limit + 1,
+            order: {
+              ...getQueryOrder(usersSortField),
+            },
+          },
+          aggs: {
+            id: {
+              terms: {
+                field: 'user.id',
+              },
+            },
+            groupId: {
+              terms: {
+                field: 'user.group.id',
+              },
+            },
+            groupName: {
+              terms: {
+                field: 'user.group.name',
+              },
+            },
+          },
+        },
+      },
+      query: {
+        bool: {
+          filter,
+          must_not: [
+            {
+              term: {
+                'event.category': 'authentication',
+              },
+            },
+          ],
+        },
+      },
+      size: 0,
+      track_total_hits: false,
```

Comment:
Should this be `true` as the others are moving towards that or is this an explicit trade-off of being `false`?

Reply:
It's the other way around. This is [turned off](https://www.elastic.co/guide/en/elasticsearch/reference/current/search-request-track-total-hits.html) for perf.